loader/fixups.c

```diff
@@ -39,16 +39,48 @@ SYS_INIT(disable_bootloader_mpu, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAU
 SYS_INIT(disable_mpu_rasr_xn, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
 #endif
 
-#if defined(CONFIG_BOARD_ARDUINO_GIGA_R1) && defined(CONFIG_VIDEO)
+
+#if defined(CONFIG_BOARD_ARDUINO_GIGA_R1)  && defined(CONFIG_INPUT_GT911_INTERRUPT)
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+
+#include <zephyr/input/input.h>
+
+// experiment to try to capture touch screen events
+void (*_giga_touch_callback)(struct input_event *evt, void *user_data) = 0;
+
+void registerGigaTouchCallback(void (*cb)(struct input_event *evt, void *user_data))
+{
+	_giga_touch_callback = cb;
+}
+
+
+void touch_event_callback(struct input_event *evt, void *user_data)
+{
+    //printk("touch_event_callback(%p %p): %p %u %u %u %d\n", evt, user_data,
+    //        evt->dev, evt->sync, evt->type, evt->code, evt->value);
+	if (_giga_touch_callback)
+		(*_giga_touch_callback)(evt, user_data);
+}
+
+static const struct device *const touch_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_touch));
+INPUT_CALLBACK_DEFINE(touch_dev, touch_event_callback, NULL);
+
+#endif
+
```

Comment:
We can make this more generic, and more useful, by passing NULL as dev. Then each library can check if it's interested in this event or not:
```C
    static const struct device *const touch_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_touch));
    if (evt->dev == touch_dev) {
```

When updating this, please add it to the end of the file and revert any other unrelated changes.

```suggestion
#if defined(CONFIG_INPUT)
#include <zephyr/kernel.h>
#include <zephyr/device.h>
#include <zephyr/input/input.h>

typedef void (*zephyr_input_callback_t)(struct input_event *evt, void *user_data);

static zephyr_input_callback_t zephyr_input_cb = NULL;

void zephyr_input_register_callback(zephyr_input_callback_t cb) {
	zephyr_input_cb = cb;
}

static void zephyr_input_callback(struct input_event *evt, void *user_data) {
	if (zephyr_input_cb) {
		zephyr_input_cb(evt, user_data);
    }
}

INPUT_CALLBACK_DEFINE(NULL, zephyr_input_callback, NULL);
#endif
```